src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -161,7 +192,9 @@ private async Task StartHostAsync(CancellationToken cancellationToken, int attem
                     DisposeRequestTrackingModule();
                 }
 
-                await _host.StartAsync(cancellationToken);
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This may result in the specialized host returning a 404 from the standby warmup calls if specialization happens immediately (which we see often during Antares deployments). It'd result in a faster startup -- but perhaps odd logs. Good tradeoff?

I also just realized that because the cancellation of this token in `RestartHostAsync()` is also inside the lock, this optimization won't happen. Will try to refactor that.